### PR TITLE
CUDA codegen handling all optional arguments

### PR DESCRIPTION
**Pull request by reguly**
To address #179

### CODE REVIEW

**Comment by reguly** [comment]
@gihanmudalige @mbuergle this is now applied to all codegens. Gihan, please run the usual tests

File: translator/c/python/op2_gen_seq.py [hunk 1]
```diff
@@ -248,7 +248,26 @@ def op2_gen_seq(master, date, consts, kernels):
...
+        for g_m in range(0,nargs):
+          if maps[g_m] == OP_MAP and (not mapinds[g_m] in k) and (not optflags[g_m]):
+            k = k + [mapinds[g_m]]
+            code('map'+str(mapinds[g_m])+'idx = arg'+str(invmapinds[inds[g_m]-1])+'.map_data[n * arg'+str(invmapinds[inds[g_m]-1])+'.map->dim + '+str(idxs[g_m])+'];')
+      #do optional ones
+      if nmaps > 0:
+        for g_m in range(0,nargs):
+          if maps[g_m] == OP_MAP and (not mapinds[g_m] in k):
+            if optflags[g_m]:
+              IF('<ARG>.opt')
```
Comment (mbuergle) [inline translator/c/python/op2_gen_seq.py:264]: @reguly is this `<ARG>` correct?

Comment (reguly) [reply translator/c/python/op2_gen_seq.py:264]: Yes, <ARG>.opt gets substituted for arg0.opt (if g_m is 0). In the sequential code generator, we do not populate an optflags bitmap

Comment (mbuergle) [reply translator/c/python/op2_gen_seq.py:264]: Ok, thanks.

**Review by gihanmudalige — APPROVED** [review]
All C apps passed. 
When using PGI compiler, there are some errors occurring, but I think these are resolved in more recent ranches.

**Pull request merged**